pkg/controller/astarte/reconcile/utils.go

```diff
@@ -390,11 +390,16 @@ func getImageForClusteredResource(defaultImageName, defaultImageTag string, reso
 	return image
 }
 
-func getDeploymentStrategyForClusteredResource(cr *apiv1alpha1.Astarte, resource apiv1alpha1.AstarteGenericClusteredResource) appsv1.DeploymentStrategy {
+func getDeploymentStrategyForClusteredResource(cr *apiv1alpha1.Astarte, resource apiv1alpha1.AstarteGenericClusteredResource, component apiv1alpha1.AstarteComponent) appsv1.DeploymentStrategy {
 	if resource.DeploymentStrategy != nil {
 		return *resource.DeploymentStrategy
 	}
-	return cr.Spec.DeploymentStrategy
+	if component == apiv1alpha1.DataUpdaterPlant && resource.DeploymentStrategy == nil {
+		return appsv1.DeploymentStrategy{
+			Type: appsv1.RecreateDeploymentStrategyType,
+		}
+	}
+	return *cr.Spec.DeploymentStrategy
```

Comment:
Wait. There is a huge problem here, I honestly have no idea what happens when DeploymentStrategy is nil, and tests are passing because tests *are* forcing `Recreate`. You probably want to have a switch statement which returns the value IF it's not nil, or one of the two strategies otherwise.